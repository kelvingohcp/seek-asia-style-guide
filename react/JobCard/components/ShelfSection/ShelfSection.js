import React from 'react';
import PropTypes from 'prop-types';
import Section from '../../../Section/Section';
import Text from '../../../Text/Text';
import Button from '../../../Button/Button';
import ButtonGroup from '../../../ButtonGroup/ButtonGroup';
import defaultLink from '../Link/Link';
import styles from './ShelfSection.less';
import classnames from 'classnames';
import { hasShelfLinks, ShelfSectionPropTypes } from '../../jobCardHelper.js';

const ShelfSection = ({ shelf = {}, LinkComponent = defaultLink, showShelfSection = false }) => {
  const { shelfLinks, tagLinks } = shelf;
  if (!shelfLinks && !tagLinks || !hasShelfLinks(shelf.shelfLinks)) {
    return null;
  }
  return (<Section className={classnames(styles.root, { [styles.showShelfSection]: showShelfSection })}>
    <div className={styles.shelfDivider} />
    {shelfLinks && shelfLinks.length &&
    <Text whispering className={styles.shelfLinksContainer}>
      {shelfLinks.map((shelfItem, i) => {
        if (shelfItem && shelfItem.items && shelfItem.items.length) {
          return (
            <div key={i}>
              {`${shelfItem.label}: `}
              {
                shelfItem.items.map((item, j) => {
                  const link = (<LinkComponent
                    link={item.link}
                    className={styles.shelfLink}
                    key={j}
                    title={item.title}>
                    {item.name}
                  </LinkComponent>);
                  if (item.children && item.children.length) {
                    return [
                      link,
                      ' > ',
                      item.children.map((child, k) => (
                        <LinkComponent
                          link={child.link}
                          className={styles.shelfLink}
                          key={`${j}${k}`}
                          title={child.title}>
                          {child.name}
                        </LinkComponent>
                      )).reduce((prev, curr) => [prev, ' | ', curr])
                    ];
                  }
                  return [link];
                }).reduce((prev, curr) => [prev, ', ', curr])
              }
            </div>);
        }
        return null;
      })}
    </Text>
    }

    {tagLinks &&
      <ButtonGroup className={styles.tagLinksContainer}>
        {tagLinks.map((item, i) => (
          <Button color="primary" compact component="a" href={item.link} className={styles.tagLink} key={i} title={item.title}>
            {item.name}
          </Button>
        ))}
      </ButtonGroup>
    }
  </Section>);
};

export default ShelfSection;
ShelfSection.propTypes = {
  shelf: ShelfSectionPropTypes,
  LinkComponent: PropTypes.func,
  showShelfSection: PropTypes.bool
};
