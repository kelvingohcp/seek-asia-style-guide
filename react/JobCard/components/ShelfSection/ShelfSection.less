@import (reference) "~seek-asia-style-guide/theme";
@import (reference) '../../JobCard.less';

.root {
  padding-top: @job-card-section-space;
  color: @saGrey2;
  display: none;
}

.showShelfSection {
  @media @desktop {
    display: block;
  }
}

.shelfDivider {
  border-top: 1px solid @saGrey4;
}

.shelfLinksContainer {
  margin: @grid-base * 2 0;
}

.shelfLink {
  .link(@actionBlue, true);
}

.tagLinksContainer {
  margin: @grid-base * 2 0;
}

.tagLink {
  line-height: @grid-base * 6;
  height: @grid-base * 6;
}
