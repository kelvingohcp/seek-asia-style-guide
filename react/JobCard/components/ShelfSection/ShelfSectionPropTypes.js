import PropTypes from 'prop-types';

export const ShelfPropTypes = PropTypes.shape({
  shelfLinks: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      link: PropTypes.string,
      title: PropTypes.string,
      children: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        link: PropTypes.string,
        title: PropTypes.string
      }))
    }))
  })),
  tagLinks: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    link: PropTypes.string,
    title: PropTypes.string
  }))
});

export const ShelfSectionPropTypes = {
  shelf: ShelfPropTypes,
  LinkComponent: PropTypes.func,
  showShelfSection: PropTypes.bool,
  trackLinkClicked: PropTypes.func
};

