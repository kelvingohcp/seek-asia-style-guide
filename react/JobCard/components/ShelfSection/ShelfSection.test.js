import React from 'react';
import { shallow } from 'enzyme';

import ShelfSection from './ShelfSection';

const defaultShelf = {
  shelfLinks: [
    {
      label: 'Job function',
      items: [
        {
          name: 'Accounting',
          link: '/jobCard',
          children: [
            {
              name: 'Accountant',
              link: '/jobCard'
            },
            {
              name: 'Audit',
              link: '/jobCard'
            },
            {
              name: 'Others',
              link: '/jobCard'
            }
          ]
        },
        {
          name: 'Admin & HR',
          link: '/jobCard',
          children: [
            {
              name: 'Receptionist',
              link: '/jobCard'
            },
            {
              name: 'Others',
              link: '/jobCard'
            }
          ]
        }
      ]
    },
    {
      label: 'Industry',
      items: [{
        name: 'Accounting / Audit / Tax Services',
        link: '/jobCard'
      }]
    }
  ],
  tagLinks: [{
    name: 'keyword 1',
    link: '/jobCard'
  }
  ]
};

/* eslint-disable react/prop-types */
const customLink = ({ link, children, ...restProps }) => (
  <span {...restProps}>{children}</span>
);

describe('JobCard - ShelfSection', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ShelfSection shelf={defaultShelf} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom link', () => {
    const wrapper = shallow(<ShelfSection shelf={defaultShelf} LinkComponent={customLink} />);
    expect(wrapper).toMatchSnapshot();
  });
});
