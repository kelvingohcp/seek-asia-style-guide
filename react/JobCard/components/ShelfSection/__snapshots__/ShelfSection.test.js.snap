// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobCard - ShelfSection should render correctly 1`] = `
<Hidden
  aboveMobile={true}
  className=""
  component="span"
  desktop={false}
  mobile={true}
  print={false}
  screen={false}
>
  <Section
    className="root"
    header={false}
    pullout={false}
    slim={false}
  >
    <div
      className="shelfDivider"
    />
    <Text
      baseline={true}
      className="shelfLinksContainer"
      whispering={true}
    >
      <div
        key="Job function"
      >
        Job function: 
        <Link
          className="shelfLink"
          key="Accounting"
          link="/jobCard"
          name="Accounting"
          onClick={[Function]}
          title="Limit result to Accounting"
        >
          Accounting
        </Link>
         &gt; 
        <Link
          className="shelfLink"
          key="Accountant"
          link="/jobCard"
          name="Accountant"
          onClick={[Function]}
          title="Limit result to Accountant in Accounting"
        >
          Accountant
        </Link>
         | 
        <Link
          className="shelfLink"
          key="Audit"
          link="/jobCard"
          name="Audit"
          onClick={[Function]}
          title="Limit result to Audit in Accounting"
        >
          Audit
        </Link>
         | 
        <Link
          className="shelfLink"
          key="Others"
          link="/jobCard"
          name="Others"
          onClick={[Function]}
          title="Limit result to Others in Accounting"
        >
          Others
        </Link>
        , 
        <Link
          className="shelfLink"
          key="Admin & HR"
          link="/jobCard"
          name="Admin & HR"
          onClick={[Function]}
          title="Limit result to Admin & HR"
        >
          Admin & HR
        </Link>
         &gt; 
        <Link
          className="shelfLink"
          key="Receptionist"
          link="/jobCard"
          name="Receptionist"
          onClick={[Function]}
          title="Limit result to Receptionist in Admin & HR"
        >
          Receptionist
        </Link>
      </div>
      <div
        key="Industry"
      >
        Industry: 
        <Link
          className="shelfLink"
          key="Accounting / Audit / Tax Services"
          link="/jobCard"
          name="Accounting / Audit / Tax Services"
          onClick={[Function]}
          title="Limit result to Accounting / Audit / Tax Services"
        >
          Accounting / Audit / Tax Services
        </Link>
      </div>
    </Text>
    <ButtonGroup
      className="tagLinksContainer"
    >
      <Button
        aria-pressed={false}
        className="tagLink"
        color="primary"
        compact={true}
        component="a"
        href="/jobCard"
        key="Limit result to Keyword 1"
        role="button"
        tabIndex={0}
        title="Limit result to Keyword 1"
      >
        keyword 1
      </Button>
      <Button
        aria-pressed={false}
        className="tagLink"
        color="primary"
        compact={true}
        component="a"
        href="/jobCard"
        key="Limit result to Keyword 2"
        role="button"
        tabIndex={0}
        title="Limit result to Keyword 2"
      >
        keyword 2
      </Button>
    </ButtonGroup>
  </Section>
</Hidden>
`;

exports[`JobCard - ShelfSection should render custom link 1`] = `
<Hidden
  aboveMobile={true}
  className=""
  component="span"
  desktop={false}
  mobile={true}
  print={false}
  screen={false}
>
  <Section
    className="root"
    header={false}
    pullout={false}
    slim={false}
  >
    <div
      className="shelfDivider"
    />
    <Text
      baseline={true}
      className="shelfLinksContainer"
      whispering={true}
    >
      <div
        key="Job function"
      >
        Job function: 
        <customLink
          className="shelfLink"
          key="Accounting"
          link="/jobCard"
          name="Accounting"
          onClick={[Function]}
          title="Limit result to Accounting"
        >
          Accounting
        </customLink>
         &gt; 
        <customLink
          className="shelfLink"
          key="Accountant"
          link="/jobCard"
          name="Accountant"
          onClick={[Function]}
          title="Limit result to Accountant in Accounting"
        >
          Accountant
        </customLink>
         | 
        <customLink
          className="shelfLink"
          key="Audit"
          link="/jobCard"
          name="Audit"
          onClick={[Function]}
          title="Limit result to Audit in Accounting"
        >
          Audit
        </customLink>
         | 
        <customLink
          className="shelfLink"
          key="Others"
          link="/jobCard"
          name="Others"
          onClick={[Function]}
          title="Limit result to Others in Accounting"
        >
          Others
        </customLink>
        , 
        <customLink
          className="shelfLink"
          key="Admin & HR"
          link="/jobCard"
          name="Admin & HR"
          onClick={[Function]}
          title="Limit result to Admin & HR"
        >
          Admin & HR
        </customLink>
         &gt; 
        <customLink
          className="shelfLink"
          key="Receptionist"
          link="/jobCard"
          name="Receptionist"
          onClick={[Function]}
          title="Limit result to Receptionist in Admin & HR"
        >
          Receptionist
        </customLink>
      </div>
      <div
        key="Industry"
      >
        Industry: 
        <customLink
          className="shelfLink"
          key="Accounting / Audit / Tax Services"
          link="/jobCard"
          name="Accounting / Audit / Tax Services"
          onClick={[Function]}
          title="Limit result to Accounting / Audit / Tax Services"
        >
          Accounting / Audit / Tax Services
        </customLink>
      </div>
    </Text>
    <ButtonGroup
      className="tagLinksContainer"
    >
      <Button
        aria-pressed={false}
        className="tagLink"
        color="primary"
        compact={true}
        component="a"
        href="/jobCard"
        key="Limit result to Keyword 1"
        role="button"
        tabIndex={0}
        title="Limit result to Keyword 1"
      >
        keyword 1
      </Button>
      <Button
        aria-pressed={false}
        className="tagLink"
        color="primary"
        compact={true}
        component="a"
        href="/jobCard"
        key="Limit result to Keyword 2"
        role="button"
        tabIndex={0}
        title="Limit result to Keyword 2"
      >
        keyword 2
      </Button>
    </ButtonGroup>
  </Section>
</Hidden>
`;
